src/auth/Crypto.h

```diff
@@ -138,9 +144,10 @@ class CryptoHandler {
   virtual int get_type() const = 0;
   virtual int create(bufferptr& secret) = 0;
   virtual int validate_secret(const bufferptr& secret) = 0;
-  virtual CryptoKeyHandler *get_key_handler(const bufferptr& secret,
-					    string& error) = 0;
+  virtual CryptoKeyHandler *get_key_handler(const bufferptr& secret, const bufferptr& iv,
+              string& error) = 0;
 
+  static const bufferptr ceph_iv;
```

Comment:
i think moving this outside of `class CryptoHandler` and giving it a more descriptive name (it should say AES) would clarify its purpose. maybe `ceph::AES_IV`? we might also consider adding something like `ceph::EMPTY_IV` for callers or algorithms that don't use one